source/includes/steps-tutorial-android-kotlin.yaml

```
@@ -0,0 +1,1323 @@
+            android:contentDescription="@string/new_task"
+            app:srcCompat="@mipmap/ic_plus"/>
+
+     </androidx.coordinatorlayout.widget.CoordinatorLayout>
+
+  To add the icon for the Floating Action Button:
+
+  1. Expand the ``app > res`` folders.
+
+  2. Right click the ``mipmap`` folder.
+
+  3. Select ``New > Image Asset``.
+
+  4. Change the "Asset Type" to "Clip Art".
+
+  5. Click the Android icon that displays.
+
+  6. In the "Select Icon" dialog box, search "add".
+
+  7. Select the icon labeled "add".
+
+  8. Click "OK".
+  
+  9. In the "Name" input, enter "ic_plus".
+
+  10. Click "Next".
+
+  11. Click "Finish".
+
+  ``TaskActivity`` also contains an options menu. To create the layout
+  for the options menu:
+  
+  1. Expand the ``app > res`` folders.
+  
+  2. Right click on the ``res`` folder.
+
+  3. Select ``New > Android Resource Directory``.
+
+  4. Use the "Resource type" dropdown to select the "menu" type.
+
+  5. Click "OK".
+
+  Once you've created the menu resource directory, right click the new
+  ``menu`` folder and select ``New > Menu Resource File``. Enter
+  "activity_task_menu" as the File name. Click "OK".
+
+  Replace the contents of ``activity_task_menu.xml`` with the following:
+
+  .. code-block:: xml
+
+     <menu xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        xmlns:tools="http://schemas.android.com/tools"
+        tools:context=".CounterActivity">
+        <item
+            android:id="@+id/action_logout"
+            android:orderInCategory="100"
+            android:title="@string/logout"
+            android:text="@string/logout"
+            app:showAsAction="always"/>
+     </menu>
+
+  These layouts uses string constants. To add them to this project,
+  expand the ``app > res > values`` folders and open ``strings.xml``.
+  Add the following lines:
+
+  .. code-block:: xml
+  
+     <string name="new_task">Create new task</string>
+     <string name="logout">Log Out</string>
+
+  Now that you've defined the widgets for the Task Activity, it's time
+  to wire those widgets up to programmatic logic.
+---
+title: Implement the Task Activity Logic
+ref: implement-the-task-activity-logic
+content: |
+  Replace the contents of ``TaskActivity`` with the following code:
+
+  .. code-block:: kotlin
+
+     package com.mongodb.tasktracker
+
+     import android.app.AlertDialog
+     import android.content.Intent
+     import android.os.Bundle
+     import android.util.Log
+     import android.view.Menu
+     import android.view.MenuItem
+     import android.widget.EditText
+     import androidx.appcompat.app.AppCompatActivity
+     import androidx.recyclerview.widget.DividerItemDecoration
+     import androidx.recyclerview.widget.LinearLayoutManager
+     import androidx.recyclerview.widget.RecyclerView
+     import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton
```

This had to be just FloatingActionButton for me
```suggestion
     import com.google.android.material.floatingactionbutton.FloatingActionButton
```

---

nice catch, I was playing around with avoiding the ``mipmap`` bit but Android unfortunately makes it extremely difficult to not use an image asset of some kind for fabs :(